### PR TITLE
Better rendering tests (#9)

### PR DESCRIPTION
**Pull request by colebillys19**
Wrote rendering tests for each component (issue #6).

I added defaultProps for the components who's smoke tests were failing due to undefined props. I tested that the tool icon buttons were clickable by checking they call appropriate functions when clicked. I passed props to components when necessary (e.g. passing `expanded={true}` to the tool components to reveal their contents). I opted not to add snapshot tests, but I'm happy to add some if deemed necessary. Note that current tests will probably have to be modified as more features continue to be implemented, and new tests written as well.

Please let me know if any of the tests seem unnecessary. I'm still getting the hang of testing and would appreciate the feedback!

### CODE REVIEW

**Comment by mmmaaatttttt** [comment]
@colebillys19 -- i'll get to this today.

File: src/components/Frame.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    global.renderWithRedux(<Frame />);
+  });
+
+  it('renders appropriate content', () => {
+    const { getByTestId } = global.renderWithRedux(<Frame imageSrc="test" />);
+    expect(getByTestId('Frame-container').firstChild.alt).toBe('current frame');
+    expect(
+      getByTestId('Frame-container').firstChild.nextSibling.textContent
+    ).toBe('▶');
+  });
```
Comment (mmmaaatttttt) [inline src/components/Frame.test.js:33]: if you've got this test, i'd also write one to demonstrate that the icon changes if `playing` is set to true.

File: src/components/Header.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+describe('<Header/>', () => {
+  it('renders without crashing', () => {
+    global.renderWithRedux(<Header />);
+  });
+
+  it('renders appropriate content', () => {
+    const { getByTestId } = global.renderWithRedux(<Header />);
+    expect(getByTestId('Header-logo').textContent).toBe('GIFsmos');
+    expect(getByTestId('Header-help').textContent.slice(0, 5)).toBe('Paste');
```
Comment (mmmaaatttttt) [inline src/components/Header.test.js]:
```suggestion
    expect(getByTestId('Header-help').textContent).toMatch('Paste');
```

using `toMatch` should also work.

File: src/components/Preview.test.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+afterEach(cleanup);
+
+describe('<Preview/>', () => {
+  it('renders without crashing', () => {
+    global.renderWithRedux(<Preview />);
+  });
+
+  it('renders appropriate content', () => {
+    // render
```
Comment (mmmaaatttttt) [inline src/components/Preview.test.js]: remove comment

File: src/components/Preview.test.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    const { getByTestId } = global.renderWithRedux(
+      <Preview
+        expanded={true}
```
Comment (mmmaaatttttt) [inline src/components/Preview.test.js]: FYI, for boolean props specifically, if you are passing in `true` you can also omit the value. In other words, you could simplify this to 

```suggestion
        expanded
```

File: src/components/Preview.test.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        previewIdx={0}
+        frames={{ 1: 'test' }}
+        frameIDs={[1]}
+      />
+    );
+    expect(getByTestId('Frame-container').firstChild.alt).toBe('current frame');
+    expect(
```
Comment (mmmaaatttttt) [inline src/components/Preview.test.js]: hmm, this expect and the one above it are already handled by another test, right? do you need them here too?

File: src/components/Sidebar.js [hunk 6]
```diff
@@ -99,4 +100,9 @@ class Sidebar extends Component {
   }
 }
 
+Sidebar.defaultProps = {
+  // gifData needs a value in order for component to render
```
Comment (mmmaaatttttt) [inline src/components/Sidebar.js]: i'd remove this comment

File: src/components/Sidebar.test.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    expect(getByTestId('SidebarButton-settings-button').firstChild.alt).toBe(
+      'settings icon'
+    );
+    // help link
+    expect(getByTestId('Sidebar-help-link').textContent).toBe('Help');
+  });
+
+  it('checks that buttons call appropriate functions when clicked', () => {
+    // mock functions
+    const requestFrame = jest.fn();
```
Comment (mmmaaatttttt) [inline src/components/Sidebar.test.js:33]: nice job on these 👍

File: src/components/SidebarButtonWithBadge.test.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  it('renders without crashing', () => {
+    global.renderWithRedux(<SidebarButtonWithBadge />);
+  });
+
+  it('renders appropriate content', () => {
+    const { getByTestId } = global.renderWithRedux(
+      <SidebarButtonWithBadge showBadge={true} color="color" />
+    );
+    // check that badge is present
+    expect(getByTestId('SidebarButton-icon-button').lastChild.tagName).toBe(
```
Comment (mmmaaatttttt) [inline src/components/SidebarButtonWithBadge.test.js]: again, i don't really care if it's a div or not -- what's the thing you're actually trying to test?

**Review by mmmaaatttttt — CHANGES_REQUESTED** [review]
@colebillys19 nice work on this. some general feedback:

1. You've got a couple of tests checking if the tagname is a DIV -- these aren't particularly interesting. Try to find some other characteristic element of the things that's rendering that you can test. (Does it have the right content? does it have the right class name? etc)

2. Another category of test to write is when you render different things based on different props passed in. This is especially true when the rendering is substantially different (e.g. rendering an error or not, depending on whether there's a message).

3. I wrote it in specific comments, but it would be nice if you didn't have to add `data-testid` attributes everywhere. Do you know whether or not this is necessary?

File: package.json [hunk 9]
```diff
@@ -5,9 +5,9 @@
   "dependencies": {
     "classnames": "^2.2.6",
     "downloadjs": "^1.4.7",
-    "react": "^16.7.0",
-    "react-dom": "^16.7.0",
-    "react-redux": "^6.0.0",
+    "react": "^16.8.6",
```
Comment (mmmaaatttttt) [inline package.json]: is there a reason you need to bump the react version?

File: src/components/ErrorToast.test.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+describe('<ErrorToast/>', () => {
+  it('renders without crashing', () => {
+    global.renderWithRedux(<ErrorToast />);
+  });
+
+  it('renders appropriate content', () => {
+    const { getByText } = global.renderWithRedux(
+      <ErrorToast message="error" />
+    );
+    expect(getByText('error').tagName).toBe('DIV');
```
[the remark this answers is not in the document]
Comment (colebillys19) [reply src/components/ErrorToast.test.js]: The idea was just to test whether the message was showing up, I rewrote it to make it more clear

File: src/components/Frame.test.js (hunk 1 above)
Comment (colebillys19) [reply src/components/Frame.test.js:33]: Done

File: src/components/Header.test.js (hunk 2 above)
Comment (colebillys19) [reply src/components/Header.test.js]: Done

File: src/components/Preview.test.js (hunk 4 above)
Comment (colebillys19) [reply src/components/Preview.test.js]: Done

File: src/components/Preview.test.js (hunk 5 above)
Comment (colebillys19) [reply src/components/Preview.test.js]: Done

File: src/components/Sidebar.js (hunk 6 above)
Comment (colebillys19) [reply src/components/Sidebar.js]: Done

File: src/components/SidebarButtonWithBadge.test.js (hunk 8 above)
Comment (colebillys19) [reply src/components/SidebarButtonWithBadge.test.js]: Done

File: package.json (hunk 9 above)
Comment (colebillys19) [reply package.json]: Noted, I'll change it back

**Pull request merged**